modules/PatternUtils.js

```diff
@@ -94,37 +94,29 @@ export function matchPattern(pattern, pathname) {
   }
 
   const match = pathname.match(new RegExp(`^${regexpSource}`, 'i'))
+  if (match == null) {
+    return null
```

Comment:
I updated all consumers of `matchPattern`. It's a lot clearer to check for a `null` response value than check for e.g. `matched.paramNames == null`.